### PR TITLE
Added a pipeline for phasing VCFs and benchmarking their performance with WhatsHap

### PR DESCRIPTION
**Pull request by rickymagner**
As the title suggests, this PR contains the following:
- A workflow for phasing a VCF using an input bam with WhatsHap
- A workflow for benchmarking a phased VCF against a baseline using WhatsHap
- Some dockers to facilitate these new workflows

More detailed documentation can be found in the /BenchmarkPhasing/README.md. 

Note I haven't done much comparison across tools yet, so future development could include allowing the choice between WhatsHap and other phasing engines, similar to our old benchmarking script. For now I wanted to collect these together just to have something.

### CODE REVIEW

File: BenchmarkPhasing/BenchmarkPhasing.wdl [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+        > full-jaccard.tsv
+
+        for i in {0..~{length(interval_beds)-1}}
+        do
+        bedtools intersect -a "${INTERVAL_BEDS[$i]}" -b ~{switch_errors_bed} > intersect.bed
+        bedtools sort -i intersect.bed > intersect-sorted.bed
+        bedtools jaccard -a intersect-sorted.bed -b ~{switch_errors_bed} > jaccard.tsv
+        tail -n 1 jaccard.tsv > contents.tsv
+        echo "${INTERVAL_LABELS[$i]}" > names.txt
+        paste names.txt contents.tsv >> full-jaccard.tsv
```
Comment (gileshall) [inline BenchmarkPhasing/BenchmarkPhasing.wdl:135]: Please indent this

File: BenchmarkPhasing/BenchmarkPhasing.wdl [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+        Array[File] interval_beds
+        Array[String] interval_bed_labels
+
+        # Runtime arguments
+        Int disk_size = 2 * ceil(size(switch_errors_bed, "GB") + size(interval_beds, "GB")) + 50
+        Int cpu = 4
+        Int memory_ram = 16
+    }
+
+    command <<<
```
Comment (gileshall) [inline BenchmarkPhasing/BenchmarkPhasing.wdl:120]: This script is complicated enough that I feel it deserves a few line blurb about what's going on, and possibly a little verbal unpacking of the various pasting and sed'n.  Props on using `yes(1)` with a command line argument, I think that's the first time I've seen that one lol.

File: BenchmarkPhasing/BenchmarkPhasing.wdl [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+        set -xue
+
+        whatshap compare \
+        --names ~{default="truth" base_sample_name},~{default="callset" call_sample_name} \
+        --tsv-pairwise eval.tsv \
+        --ignore-sample-name \
+        ~{true="--only-snvs" false="" only_snvs} \
+        --switch-error-bed switch_errors.bed \
+        --longest-block-tsv longest_blocks.tsv \
+        ~{base_vcf} ~{call_vcf}
```
Comment (gileshall) [inline BenchmarkPhasing/BenchmarkPhasing.wdl:79]: Please indent

File: Utilities/Dockers/Bedtools/Dockerfile [hunk 4]
```diff
@@ -0,0 +1,10 @@
+FROM ubuntu:22.10
+
+RUN apt update
+RUN apt install -y wget
+RUN wget https://github.com/arq5x/bedtools2/releases/download/v2.30.0/bedtools.static.binary
+RUN mv bedtools.static.binary bedtools
+RUN chmod a+x bedtools
+RUN mv bedtools /root/bedtools
```
[the remark this answers is not in the document]
Comment (rickymagner) [reply Utilities/Dockers/Bedtools/Dockerfile]: Nice, I didn't know about `ADD` before, this is much better!

File: Utilities/Dockers/WhatsHap/Dockerfile [hunk 5]
```diff
@@ -0,0 +1,7 @@
+FROM ubuntu:23.04
+RUN apt update
+RUN apt install -y git python3 python3-pip python3-dev build-essential libbz2-dev liblzma-dev libcurl4-openssl-dev libncurses-dev tabix
+
+RUN pip install whatshap==1.7
```
[the remark this answers is not in the document]
Comment (rickymagner) [reply Utilities/Dockers/WhatsHap/Dockerfile]: Good call. I mistakenly thought `apt-get` was getting deprecated in favor of `apt` some years ago and never went back. In a script like this it makes sense to use `apt-get`

File: BenchmarkPhasing/BenchmarkPhasing.wdl [hunk 6]
```diff
@@ -0,0 +1,162 @@
...
+        Int memory_ram = 16
+    }
+
+    command <<<
+        set -xue
+
+        INTERVAL_BEDS=(~{sep=' ' interval_beds})
+        INTERVAL_LABELS=(~{sep=' ' interval_bed_labels})
+
+        > full-jaccard.tsv
```
[the remark this answers is not in the document]
Comment (rickymagner) [reply BenchmarkPhasing/BenchmarkPhasing.wdl]: Oops, for some reason there I thought the file needed to be created before appending to it later. Removed this line

**Pull request merged**